advanced/src/hashTable.js

```diff
@@ -0,0 +1,227 @@
+function Node(key, value) {
+  this.key = key
+  this.value = value
+  this.next = null
+}
+
+
+const LinkedList = class LinkedList {
+  constructor() {
+    this.length = 0
+    this.head = null
+  }
+
+  insert(key, value) {
```

Comment:
Since you're using this `LinkedList` in a `HashTable`, you probably want to optimize by doing inserts at the head of the list - this way we minimize the amount of time `put` takes.  This *may* affect retrieval time, but it really depends on the distribution of data in buckets, whether or not we frequently retrieve the most recently inserted items or the least recently inserted items, etc.